bundles/org.openhab.binding.bluetooth/src/main/java/org/openhab/binding/bluetooth/discovery/internal/BluetoothDiscoveryService.java

```diff
@@ -153,7 +153,9 @@ public void deviceDiscovered(BluetoothDevice device) {
         logger.debug("Discovered bluetooth device '{}': {}", device.getName(), device);
 
         DiscoveryCache cache = discoveryCaches.computeIfAbsent(device.getAddress(), addr -> new DiscoveryCache());
-        cache.handleDiscovery(device);
+        if (cache != null) {
+            cache.handleDiscovery(device);
+        }
```

Comment:
Do you agree that cache will never be null ?
Your change is just to avoid a wrong compiler analysis ?

Reply:
I considered to add a debug log entry if it is null, but i'm pretty sure cache will never be null here.

Reply:
So what was the reason to add a null test ?

Reply:
The compiler warns about it and i'm not 100% sure that it is never null. So to prevent a possible NPE i added the check. If you want it removed, i;m fine with that.

Reply:
Let's keep it.

Reply:
@lolodomo - you are right, it cannot be null. The analysis doesn't understand `computeIfAbsent`. One approach could be:
```java
        DiscoveryCache cache = Objects
                .requireNonNull(discoveryCaches.computeIfAbsent(device.getAddress(), addr -> new DiscoveryCache()));
```

This would still check for null, but instead of skipping logic, it will throw `NullPointerException`. Neither can actually happen, but I think it would make it more clear what is expected here.

Or even more explicit, but slightly more boiler-plate:
```java
        DiscoveryCache cache = discoveryCaches.computeIfAbsent(device.getAddress(), addr -> new DiscoveryCache());
        if (cache != null) {
            throw new IllegalStateException("DiscoveryCache is null");
        }
        cache.handleDiscovery(device);
```

Reply:
Your proposal to use Objects.requireNonNull looks good to me. Maybe you can adjust that @lsiepel ?

Reply:
Used Objects.requireNonNull on several other PR's lately, This is a perfect fix for this case. Thanks for the suggestion.